src/rpm/builder.rs

```diff
@@ -144,17 +140,20 @@ impl RPMBuilder {
     ) -> Result<(), RPMError> {
         let dest = options.destination;
         if !dest.starts_with("./") && !dest.starts_with('/') {
-            return Err(RPMError::new(&format!(
-                "invalid path {} - needs to start with / or ./",
-                dest
-            )));
+            return Err(RPMError::InvalidDestinationPath {
+                path: dest,
+                desc: "needs to start with / or ./",
+            });
         }
 
         let pb = PathBuf::from(dest.clone());
 
         let parent = pb
             .parent()
-            .ok_or_else(|| RPMError::new(&format!("invalid destination path {}", dest)))?;
+            .ok_or_else(|| RPMError::InvalidDestinationPath {
+                path: dest.clone(),
+                desc: "does not have a parent directory",
```

Comment:
We should consider phrasing those in a consistent way (capitalization, active/passive form).